### PR TITLE
added schemas

### PR DESCRIPTION
**Pull request by dmaitha**
Hey guys take a look at the new schemas/models I made.  Try and see if they make sense for the features you need to query.

The Task/User join is a big one for the site it will:
1. Assign a badge to a user.  A user can have many badges assigned to him but not the same badge.
2. Award the user.  When the user has completed all the tasks for the badge, i.e. the task_completed array is full the status of awarded will be true.
3. Particularly for the admin page, I'd imagine you could get easily get the best teams by counting which teamId has the highest amount of awarded=true.  You could also get most awarded badges, least awarded etc.  Getting the highest performing users could also be done with this collection.

I'll make the seed script for this.  If we go forward with this design then when badges are created we will have to populate two collection, the badge and the tasks appropriately.

https://docs.google.com/presentation/d/1gryfrWxxnA6tDf3YkDr2wcUpVPjLqCKbqFbG_lU3wvI/edit#slide=id.g64effd68a3_3_0

### CODE REVIEW

File: server/models/Tasks.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  Description: {
+    type: String
+  },
+  Badge: {
+    type: Schema.Types.ObjectId,
+    ref: "badges"
+  }
+});
+
+const taskModel = mongoose.model("tasks", taskSchema, "tasks");
```
Comment (GauravDeshpande) [inline server/models/Tasks.js]: Why do you have the model written as "tasks", taskSchema, "tasks"? why is "tasks" written twice"

File: server/models/TaskUserJoin.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+  ]
+});
+
+const taskModel = mongoose.model(
+  "taskUserJoin",
+  taskUserJoinSchema,
+  "tasksUserJoin"
+);
+
```
Comment (GauravDeshpande) [inline server/models/TaskUserJoin.js]: Where is "taskUserJoin" and "tasksUserJoin" coming from in this instance?

File: server/models/Badge.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  Team: {
+    type: Schema.Types.ObjectId,
+    ref: "teams"
+  },
+  Recipients: [{ type: Schema.Types.ObjectId, ref: "users" }],
+  Tasks: [{ type: Schema.Types.ObjectId, ref: "tasks" }]
+});
+
+const badgeModel = mongoose.model("badges", badgeSchema, "badges");
+
```
Comment (GauravDeshpande) [inline server/models/Badge.js]: Why is badges used as reference twice?

File: server/models/Tasks.js (hunk 1 above)
Comment (dmaitha) [reply server/models/Tasks.js]: I think we can actually get rid of this.  It just a third argument to the Mongoose.model I found when I was having trouble doing the joins.  I dont think we need it.

**Pull request closed without merging**